scripts/gn/main.py

```diff
@@ -0,0 +1,209 @@
+        sys: {str(s) for s in signals if s.startswith(PHASE_PREFIX)} for sys, signals in sys_signals.items() if signals
+    }
+    return filtered_sys_signals
+
+
+def ensure_folders(paths: [Path]) -> None:
+    """
+    Ensures the list of directories exist in the file system.
+
+    :param Directories to create if they do not already exist
+    """
+    for path in paths:
+        if not path.is_dir():
+            path.mkdir(parents=True)
+
+
+@click.group()
+@click.option(
+    "--log-level",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]),
+    default="INFO",
+    help="Set the logging level.",
+)
+def cli(log_level):
+    """A collection of tools to prepare Ginan for processing"""
+    # Set up logging based on the provided log level
+    root_logger = logging.getLogger()
+    root_logger.setLevel(getattr(logging, log_level))
+
+
+@cli.command()
+@click.option("--ppp", is_flag=True, help="Use Precise Point Positioning (PPP).")
+@click.option("--static", is_flag=True, help="Use static mode.")
+@click.option("--rinex-path", type=click.Path(exists=True), help="Path to RINEX file.")
+@click.option(
+    "--workspace-root", type=click.Path(), help="Path to store workspaces", default="workspace", required=False
+)
+@click.option("--workspace-name", type=str, help="Name for the workspace", default="network", required=False)
+def prep(ppp: bool, static: bool, rinex_path: Path, workspace_root: Path, workspace_name: str):
+    """
+    Downloads required IGS products and generates template YAML
+    """
+    if ppp and static:
+        rinex_path = Path(rinex_path)
+        target_dir = Path(workspace_root) / workspace_name
+        download_dir = target_dir / "downloads"
+        data_dir = target_dir / "data"
+        ensure_folders([target_dir, download_dir, data_dir])
+
+        # Make a copy of the rinex file in the target directory
+        new_rinex_path = data_dir / f"{rinex_path.stem}.rnx"
+        shutil.copy(rinex_path, new_rinex_path)
+
+        logging.info("Parsing rinex file")
+        header = parse_v3_header(rinex_path)
+
+        # TODO: The pea does not allow station names of 4 digits eg. 7369 - but it would be good if it did.
+        # For now, create an alias and rename the copy
+        station_alias = header.get_station_alias()
+        shutil.move(new_rinex_path, data_dir / f"{station_alias}.rnx")
+
+        logging.info("Downloading required files from IGS CDDIS...")
+        download(header, download_dir)
+        logging.info("Downloads complete")
+
+        logging.info("Generating yaml config file")
+        overrides = create_overrides(header, station_alias, workspace_name)
+        yaml_path = write_yaml(target_dir, workspace_name=workspace_name, overrides=overrides)
+
+        logging.info("Ready to run the pea...")
+        return yaml_path
+    else:
+        raise NotImplementedError(
+            "Only static ppp is supported through this interface - include --static --ppp"
```

Comment:
```suggestion
            "Only static ppp is supported through this interface - include --static --ppp "
```
Will concatenate to `-pppYou` as is